manifests/base/templates/endpoints.yaml

```diff
@@ -1,3 +1,37 @@
+{{- if .Values.global.remotePilotAddress }}
+apiVersion: v1
+kind: Endpoints
+metadata:
+  name: istio-pilot
```

Comment:
is istio-pilot endpoint really needed?

Reply:
It would only be needed if we need to support multicluster with the pre-istiod deployment model.

Reply:
ok SG

Reply:
BTW - I think the istiod & istio-pilot endpoints will also be created as the result of pilot.enabled=true.  so do we still need to create this again?

Reply:
`istio-pilot` shouldn't be needed in 1.6+. I kept it because it would make backporting to 1.5 a bit easier. If someone was using the legacy charts and separate services in 1.5 they wouldn't be enabling pilot in the remote cluster.

Reply:
isn't enabling pilot the only way to enable istiod?  That is what the remote profile has today.

Reply:
istiod runs in each clsuter and we need the istiod-remote service to avoid conflicting the "real" istiod service in the cluster.

When istiod=false (separate services) pilot doesn't need to run in the remote cluster. The `istio-pilot` service in the remote cluster points to the pilot in the main cluster. This is how things worked in <= 1.4.

Reply:
We probably don't need to support the istiod=false case for multicluster

Reply:
I agree for not needing this in master. What about 1.5?

Reply:
@ayj I'd recommend sorting out what this should look like for `1.5` in master. Then follow up with another PR after we release 1.5.1.

Reply:
Master doesn't have the legacy helm charts. We'll need to manually cherry-pick this PR anyway.